url49584c1d41_Arrem_GGS_tgz-pJ8-com_gamezgalaxy_test_console_MainJ8/src/com/gamezgalaxy/GGS/world/blocks/Lava.java

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+ * Copyright (c) 2012 GamezGalaxy.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Public License v3.0
+ * which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/gpl.html
+ ******************************************************************************/
+package com.gamezgalaxy.GGS.world.blocks;
+
+import java.util.Random;
+
+import com.gamezgalaxy.GGS.server.Server;
+import com.gamezgalaxy.GGS.world.Block;
+import com.gamezgalaxy.GGS.world.PhysicsBlock;
+
+public class Lava extends PhysicsBlock {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+	
+	private int time;
+	
+	private int type;
+	
+	private final Random random = new Random();
+
+	public Lava(byte ID, String name) {
+		super(ID, name);
+	}
+	
+	public Lava() {
+		super((byte)10, "Lava");
+	}
+
+	public Lava(byte b, String string, Server s) {
+		super(b, string, s);
+	}
+
+	@Override
+	public PhysicsBlock clone(Server s) {
+		Lava l = new Lava((byte)10, "Lava", s);
+		l.type = type;
+		if (l.type == 0)
+			l.type = random.nextInt(6) + 1;
+		return l;
+	}
+
+	@Override
+	public void tick() {
+		if (!nearSponge(getLevel().PosToInt(getX(), getY(), getZ()))) {
+			if (time < random.nextInt(6)) {
+				if (type == 2 && check(getLevel().PosToInt(getX(), getY() - 1, getZ())))
+					add(getX(), getY() - 1, getZ());
+				time++;
+				return;
+			}
+			int x1 = random.nextInt(3);
+			int x2 = random.nextInt(3);
+			int z1 = random.nextInt(3);
+			int z2 = random.nextInt(3);
+			int i = 0;
+			while (i != x1 && (getLevel().getTile(getX() + i + 1, getY(), getZ()).getVisableBlock() == 0 || getLevel().getTile(getX() + i + 1, getY(), getZ()).getVisableBlock() == this.getVisableBlock())) {
+				i++;
+				if (!check(getLevel().PosToInt(getX() + i, getY(), getZ())))
+					break;
+				else
+					add(getX() + i, getY(), getZ());
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 64:70, 80:86.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.